### ISSUE TITLE
Add missing data

### ISSUE THREAD

**Issue by sniffertine** [datasets/unece-units-of-measure#1]
Dear developer

I was looking for such a dataset for quite a while any very happy that I found it here. When I was using it, I saw that some data has been stripped out (symbol, conversion factor, deprecated, deleted)

Also I am not quite sure what revision of the Excel source was used to generate the dataset. 

Would it be a huge effort to add the missing data? 
For example:
- add column for symbol
- add column for conversion factor
- add column for deprecated (true/false or even the date of deprecation)
- add column for deleted (true/false or even the date of deletion)

**Comment by sabas** [comment]
Hello, I used the excel from the unece website if I recall
https://www.unece.org/uncefact/codelistrecs.html
But it was a mess to convert to an usable version and I simply dropped the columns I couldn't convert.
The revision I used is the 13th (I wrote only in the json description). It could be integrated in the README.

Would you like to try and upgrade this dataset? I see that they published the 14th revision.

**Comment by sniffertine** [comment]
Hi

I could imagine to contribute here. How did you generate the dataset? I would automate this using a small java program that processes the excel. 

Also, I would like to add a build gradle file to the dataset root. If present, (Java/Android/JVM) developers could declare the dataset as a compile time dependency using https://jitpack.io/

**Comment by sabas** [comment]
In the other package I contributed I added the script in a ```scripts``` folder
https://github.com/datasets/un-locode

For the gradle file, I am not sure, @rufuspollock @Mikanebu could you assist?

**Comment by sniffertine** [comment]
So for this package you do not have the scripts anymore?

I could build the gradle file.

**Comment by sabas** [comment]
Sorry forgot to reply on that point, I built the package manually :D

**Comment by rufuspollock** [comment]
@sniffertine we'd welcome patches / improvements. We'd love to have a gradle file - and a generic process for providing this.

@nirabpudasaini @lauragift21  can you keep an eye here and help out.

**Comment by sniffertine** [comment]
I am now starting to create a small java program to extract the data from the excel file automatically. 

In the README, @sabas said that he `merged the two Annexes`. I wonder how you handled duplicate entries for some common codes have duplicates (for example `C62` has 95 entries in Annex I and one entry in Annex II / III). It seems you just used the first entry of Annex I

**Comment by sniffertine** [comment]
As I understand it while reading the `rec20_Rev7e_2010.pdf`, I probably could just use Annex III as it contains all level 1, 2 and 3 entries listed by common code

**Comment by sabas** [comment]
Yeah I checked the recommendation again and Annex II and III should be the same but sorted differently. Perhaps I used the two together because the columns were formatted differently? I don't rememender :(

**Comment by sniffertine** [comment]
As I said, some codes have multiple entries in Annex I.

**Comment by sniffertine** [comment]
I have now implemented automatic data extraction of the excel, based on the annex II and III workbook. 

You can have a look here: https://github.com/sniffertine/unece-units-of-measure

Should I create a merge request?

**Comment by sabas** [comment]
Super. I think that the folder should be named scripts instead of data-generation per the data packages specification (https://specs.frictionlessdata.io/data-package/).

**Comment by sniffertine** [comment]
Here is the current development branch: https://github.com/sniffertine/unece-units-of-measure/tree/feature/Rec20rev14e-Annex_II-III_2020

**Comment by tstorek** [comment]
Hello, what is the current state of this?

**Comment by sniffertine** [comment]
My pull request is still open: https://github.com/datasets/unece-units-of-measure/pull/3

**Comment by sabas** [comment]
I merged it, perhaps last year I missed the email.
Strange times they were :-)

**Comment by sniffertine** [comment]
nice :)

could you tag the current state as `rec20_Rev14e_2020`?

**Comment by tstorek** [comment]
Hey,

thank you very much. @sabas @sniffertine is there a reason not to add the quantity column? That would help me a lot.
Alternatively, I need to download the original excel table :/

**Comment by sabas** [comment]
@sniffertine done

**Comment by tstorek** [comment]
Thank you!

**Comment by sniffertine** [comment]
@tstorek the quantity comes from Annex I, I used the data from Annex II and III. If you explain me how those Annexes work together. If you read above, I had issues with duplicates when processing Annex I

**Comment by tstorek** [comment]
@sniffertine I checked the annexes. Maybe we can merge the two of them. I would suggest the following:
remove all units with status X or D --> since the remainder of Annexes 2 & 3 is longer than the one of Annex 1 I suggest using the codes from Annex 2&3 for a lookup in Annex1 and add the missing data for the codes where it is available.

But probably one does not need Level 3 units according to the information below.
Hence, only Annex1 will be sufficient. Additional information is probably never required. 
Therefore, merging would be a valid assumption:)

What do you think?

I found this information [here](https://unece.org/trade/cefact/UNLOCODE-Download):

Level 1 - normative = SI normative units, standard and commonly used multiples
NOTE: standard multiples are identified with "S" and commonly used multiples with "M"
e.g. "1 metre", "1S centimetre", "1M hectometre").

Level 2 – normative equivalent = SI normative equivalent units (UK, US, etc.) and
commonly used multiples.

Level 3 – informative = 9 categories of informative units (units of count and other
miscellaneous units), invariably with no comprehensive conversion factor to SI.
These units are provided for information and to facilitate the assignment and usage of
a common code value to represent such units.


The three annexes are structured accordingly:
Annex I – Code elements listed by quantity category. This annex is normative and
contains only level 1 and 2 entries. The columns represented in this annex are:
- Quantity
- Status
- Level/category
- Name
- Description
- Representation symbol
- Conversion factor to SI
- Common Code

Annex II – Code elements listed by name. This annex is informative and contains all
level 1, 2 and 3 entries. The columns represented in this annex are:
- Status
- Name
- Description
- Level/category
- Representation symbol
- Conversion factor to SI
- Common Code

Annex III – Code elements listed by common code. This annex is informative and
contains all level 1, 2 and 3 entries. The columns represented in this annex are:
- Status
- Common Code
- Name
- Description
- Level/category
- Representation symbol
- Conversion factor to SI.
